app/controllers/actions/DataRequiredAction.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import com.google.inject.{ImplementedBy, Inject}
+import models.requests.{DataRequest, OptionalDataRequest}
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{ActionRefiner, Result}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DataRequiredActionImpl @Inject()()(implicit val executionContext: ExecutionContext) extends DataRequiredAction {
+
+  override protected def refine[A](request: OptionalDataRequest[A]): Future[Either[Result, DataRequest[A]]] = {
+    request.userAnswers match {
+      case None => Future.successful(Left(Redirect(controllers.routes.CharitableEligibilityController.onPageLoad())))
+      case Some(data) => Future.successful(Right(DataRequest(request.request, request.sessionId, data)))
+    }
+  }
+}
+
+@ImplementedBy(classOf[DataRequiredActionImpl])
+trait DataRequiredAction extends ActionRefiner[OptionalDataRequest, DataRequest]
```

Comment:
Can the ActionRefiner be extended by the class directly so we don't need to do this ImplementedBy setup?